src/bugsnag/client.ts

```diff
@@ -745,22 +772,20 @@ export class BugsnagClient implements Client {
           "Look for display_id field in the response - these are the field names to use in filters",
         ],
       },
-      async (_args, _extra) => {
-        const projectFields = this.cache?.get<EventField[]>(
-          cacheKeys.CURRENT_PROJECT_EVENT_FILTERS,
+      async (args, _extra) => {
+        const params = listProjectEventFiltersInputSchema.parse(args);
+        const eventFilters = await this.getProjectEventFilters(
+          await this.getInputProject(params.projectId),
         );
-        if (!projectFields)
-          throw new ToolError("No event filters found in cache.");
-
         return {
-          content: [{ type: "text", text: JSON.stringify(projectFields) }],
+          content: [{ type: "text", text: JSON.stringify(eventFilters) }],
         };
       },
     );
 
     const updateErrorInputSchema = z.object({
       projectId: this.projectApiKey
-        ? toolInputParameters.projectId.optional()
+        ? toolInputParameters.projectId
         : toolInputParameters.projectId,
```

Comment:
unnecessary ternary operation

```suggestion
projectId: toolInputParameters.projectId,
```